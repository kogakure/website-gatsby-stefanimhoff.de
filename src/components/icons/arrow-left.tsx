import React, { SVGProps } from 'react';

const SvgArrowLeft = (props: SVGProps<SVGSVGElement>) => (
  <svg viewBox="0 0 24 24" width="1em" height="1em" {...props}>
    <path fill="none" d="M0 0h24v24H0z" />
    <path d="M7.828 11H20v2H7.828l5.364 5.364-1.414 1.414L4 12l7.778-7.778 1.414 1.414z" />
  </svg>
);

export default SvgArrowLeft;
