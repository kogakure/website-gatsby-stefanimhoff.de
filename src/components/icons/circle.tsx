import React, { SVGProps } from 'react';

const SvgCircle = (props: SVGProps<SVGSVGElement>) => (
  <svg
    viewBox="0 0 24 24"
    fillRule="evenodd"
    clipRule="evenodd"
    strokeLinejoin="round"
    strokeMiterlimit={2}
    width="1em"
    height="1em"
    {...props}
  >
    <circle cx={12} cy={12} r={9.967} />
  </svg>
);

export default SvgCircle;
