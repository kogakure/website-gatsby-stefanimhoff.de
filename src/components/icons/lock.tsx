import React, { SVGProps } from 'react';

const SvgLock = (props: SVGProps<SVGSVGElement>) => (
  <svg viewBox="0 0 24 24" width="1em" height="1em" {...props}>
    <path fill="none" d="M0 0h24v24H0z" />
    <path d="M19 10h1a1 1 0 011 1v10a1 1 0 01-1 1H4a1 1 0 01-1-1V11a1 1 0 011-1h1V9a7 7 0 1114 0v1zM5 12v8h14v-8H5zm6 2h2v4h-2v-4zm6-4V9A5 5 0 007 9v1h10z" />
  </svg>
);

export default SvgLock;
