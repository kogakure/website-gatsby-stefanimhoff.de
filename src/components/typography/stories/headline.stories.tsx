import React from 'react';

import { Headline } from '..';

export default {
  title: 'Typography/Headline',
};

export const Default = () => <Headline>Headline</Headline>;

export const NoMargin = () => <Headline noMargin>Headline</Headline>;
