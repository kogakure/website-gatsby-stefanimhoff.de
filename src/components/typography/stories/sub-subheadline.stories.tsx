import React from 'react';

import { SubSubheadline } from '..';

export default {
  title: 'Typography/SubSubheadline',
};

export const Default = () => <SubSubheadline>SubSubheadline</SubSubheadline>;

export const NoMargin = () => (
  <SubSubheadline noMargin>SubSubheadline</SubSubheadline>
);
