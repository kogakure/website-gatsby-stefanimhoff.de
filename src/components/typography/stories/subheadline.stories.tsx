import React from 'react';

import { Subheadline } from '..';

export default {
  title: 'Typography/Subheadline',
};

export const Default = () => <Subheadline>Subheadline</Subheadline>;

export const NoMargin = () => <Subheadline noMargin>Subheadline</Subheadline>;
