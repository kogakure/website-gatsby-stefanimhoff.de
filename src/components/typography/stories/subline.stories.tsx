import React from 'react';

import { Subline } from '..';

export default {
  title: 'Typography/Subline',
};

export const Default = () => <Subline>Subline</Subline>;
