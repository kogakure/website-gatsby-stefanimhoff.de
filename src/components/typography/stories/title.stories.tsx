import React from 'react';

import { Title } from '..';

export default {
  title: 'Typography/Title',
};

export const Default = () => <Title>Title</Title>;

export const NoMargin = () => <Title noMargin>Title</Title>;
