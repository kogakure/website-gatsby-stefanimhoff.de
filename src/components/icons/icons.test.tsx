import React from 'react';

import { render } from '../../services/test-utils';

import { Icon as ArrowUp } from './arrow-up.stories';
import { Icon as ArrowLeft } from './arrow-left.stories';
import { Icon as Circle } from './circle.stories';
import { Icon as Information } from './information.stories';
import { Icon as Lock } from './lock.stories';

describe('Icon', () => {
  test('ArrowUp renders correctly', () => {
    const { container } = render(<ArrowUp />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('ArrowLeft renders correctly', () => {
    const { container } = render(<ArrowLeft />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('Circle renders correctly', () => {
    const { container } = render(<Circle />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('Information renders correctly', () => {
    const { container } = render(<Information />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('Lock renders correctly', () => {
    const { container } = render(<Lock />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
