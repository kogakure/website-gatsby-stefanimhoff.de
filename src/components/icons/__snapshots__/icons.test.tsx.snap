// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon ArrowLeft renders correctly 1`] = `
<svg
  height="1em"
  viewBox="0 0 24 24"
  width="1em"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M7.828 11H20v2H7.828l5.364 5.364-1.414 1.414L4 12l7.778-7.778 1.414 1.414z"
  />
</svg>
`;

exports[`Icon ArrowUp renders correctly 1`] = `
<svg
  height="1em"
  viewBox="0 0 24 24"
  width="1em"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M13 7.828V20h-2V7.828l-5.364 5.364-1.414-1.414L12 4l7.778 7.778-1.414 1.414L13 7.828z"
  />
</svg>
`;

exports[`Icon Circle renders correctly 1`] = `
<svg
  clip-rule="evenodd"
  fill-rule="evenodd"
  height="1em"
  stroke-linejoin="round"
  stroke-miterlimit="2"
  viewBox="0 0 24 24"
  width="1em"
>
  <circle
    cx="12"
    cy="12"
    r="9.967"
  />
</svg>
`;

exports[`Icon Information renders correctly 1`] = `
<svg
  height="1em"
  viewBox="0 0 24 24"
  width="1em"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 100-16 8 8 0 000 16zM11 7h2v2h-2V7zm0 4h2v6h-2v-6z"
  />
</svg>
`;

exports[`Icon Lock renders correctly 1`] = `
<svg
  height="1em"
  viewBox="0 0 24 24"
  width="1em"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M19 10h1a1 1 0 011 1v10a1 1 0 01-1 1H4a1 1 0 01-1-1V11a1 1 0 011-1h1V9a7 7 0 1114 0v1zM5 12v8h14v-8H5zm6 2h2v4h-2v-4zm6-4V9A5 5 0 007 9v1h10z"
  />
</svg>
`;
